### ISSUE TITLE
Decide and specify what happens on integer overflow

### ISSUE THREAD

**Issue by Marwes** [gluon-lang/gluon#650]
Currently the raw rust operations are used for arithmetic so gluon's vm will (rust) panic on overflow in debug mode and wrap in release.

Leaning towards always checking for overflow and doing a gluon panic when it happens, it is already interpreted so the overhead shouldn't matter

**Comment by Laegluin** [comment]
> Leaning towards always checking for overflow and doing a gluon panic when it happens, it is already interpreted so the overhead shouldn't matter  
It's probably a good idea to expose rust's wrapping/saturating arithmetic functions,

**Comment by Marwes** [comment]
Yeah, that's a good idea. Opened an issue https://github.com/gluon-lang/gluon/issues/651
